st2common/bin/st2-self-check

```diff
@@ -140,7 +140,7 @@ do
 
     START_TS=$(date +%s)
     OUTPUT=$(st2 run ${TEST} protocol=${PROTOCOL} token=${ST2_AUTH_TOKEN})
-    echo ${OUTPUT} | grep "status" | grep -q "succeeded"
+    echo "${OUTPUT}" | grep "status" | grep -q "succeeded"
```

Comment:
OK, the deeper root cause found and better way after looking even deeper.

The following part:
```
OUTPUT=$(st2 run ${TEST} protocol=${PROTOCOL} token=${ST2_AUTH_TOKEN})
echo ${OUTPUT}
```
results in an entire output being on a single line due to command substitution. This is why it breaks the grep rule at the first place.

Just adding the quotes fixes it all.

Reply:
Nicely done :+1:

Reply:
Wrong: 
```bash
vagrant@stackstorm:~$ OUTPUT=$(st2 execution get 61b26b8f1838bc652ac0cef0)

vagrant@stackstorm:~$ echo $OUTPUT 
id: 61b26b8f1838bc652ac0cef0 action.ref: tests.test_inquiry_chain parameters: protocol: http token: 20e423a0e2c0442da535696a07d112cf status: failed (33s elapsed) result_task: get_workflow_details_3 result: failed: true return_code: 1 stderr: "Traceback (most recent call last): File "/opt/stackstorm/st2/lib/python3.6/site-packages/st2client/commands/resource.py", line 199, in get_resource_by_pk instance = self.manager.get_by_id(pk, **kwargs) File "/opt/stackstorm/st2/lib/python3.6/site-packages/st2client/models/core.py", line 42, in decorate return func(*args, **kwargs) File "/opt/stackstorm/st2/lib/python3.6/site-packages/st2client/models/core.py", line 238, in get_by_id self.handle_error(response) File "/opt/stackstorm/st2/lib/python3.6/site-packages/st2client/models/core.py", line 195, in handle_error response.raise_for_status() File "/opt/stackstorm/st2/lib/python3.6/site-packages/requests/models.py", line 943, in raise_for_status raise HTTPError(http_error_msg, response=self) requests.exceptions.HTTPError: 500 Server Error: Internal Server Error MESSAGE: Internal Server Error for url: http://127.0.0.1:9101/executions/61b26b901838bc652ac0cef3?include_attributes=action.ref%2Caction.runner_type%2Cstart_timestamp%2Cend_timestamp%2Clog%2Cid%2Caction.ref%2Ccontext.user%2Cparameters%2Cstatus%2Cstart_timestamp%2Cend_timestamp%2Clog%2Cresult" stdout: 'Action Execution "61b26b901838bc652ac0cef3" is not found. ERROR: Resource 61b26b901838bc652ac0cef3 not found. ' succeeded: false start_timestamp: Thu, 09 Dec 2021 20:48:15 UTC end_timestamp: Thu, 09 Dec 2021 20:48:48 UTC log: - status: requested timestamp: '2021-12-09T20:48:15.013000Z' - status: scheduled timestamp: '2021-12-09T20:48:15.122000Z' - status: running timestamp: '2021-12-09T20:48:15.357000Z' - status: failed timestamp: '2021-12-09T20:48:48.893000Z' +--------------------------+------------------------+---------------------------------+------------+-------------------------------+ | id | status | task | action | start_timestamp | +--------------------------+------------------------+---------------------------------+------------+-------------------------------+ | 61b26b8fbcc6a7ed46a5b288 | succeeded (3s elapsed) | execute_inquiry_workflow | core.local | Thu, 09 Dec 2021 20:48:15 UTC | | 61b26b93bcc6a7ed46a5b28a | succeeded (1s elapsed) | get_inquiry_trigger | core.local | Thu, 09 Dec 2021 20:48:19 UTC | | 61b26b94bcc6a7ed46a5b28c | succeeded (1s elapsed) | get_inquiry_id | core.local | Thu, 09 Dec 2021 20:48:20 UTC | | 61b26b95bcc6a7ed46a5b28e | succeeded (1s elapsed) | get_workflow_id | core.local | Thu, 09 Dec 2021 20:48:21 UTC | | 61b26b96bcc6a7ed46a5b290 | succeeded (2s elapsed) | get_workflow_details_1 | core.local | Thu, 09 Dec 2021 20:48:22 UTC | | 61b26b98bcc6a7ed46a5b292 | failed (4s elapsed) | invalid_response_expect_failure | core.local | Thu, 09 Dec 2021 20:48:24 UTC | | 61b26b9dbcc6a7ed46a5b294 | succeeded (5s elapsed) | pause_after_invalid_response | core.pause | Thu, 09 Dec 2021 20:48:29 UTC | | 61b26ba3bcc6a7ed46a5b296 | succeeded (1s elapsed) | get_workflow_details_2 | core.local | Thu, 09 Dec 2021 20:48:35 UTC | | 61b26ba5bcc6a7ed46a5b2d1 | succeeded (4s elapsed) | valid_response | core.local | Thu, 09 Dec 2021 20:48:37 UTC | | 61b26ba9bcc6a7ed46a5b2d3 | succeeded (6s elapsed) | pause_after_valid_response | core.pause | Thu, 09 Dec 2021 20:48:41 UTC | | 61b26bafbcc6a7ed46a5b2d5 | failed (1s elapsed) | get_workflow_details_3 | core.local | Thu, 09 Dec 2021 20:48:47 UTC | +--------------------------+------------------------+---------------------------------+------------+-------------------------------+

vagrant@stackstorm:~$ echo ${OUTPUT} | grep "status" | grep -q "succeeded"
echo $?
0
```

Fixed: 
```
vagrant@stackstorm:~$ echo "$OUTPUT"
id: 61b26b8f1838bc652ac0cef0
action.ref: tests.test_inquiry_chain
parameters: 
  protocol: http
  token: 20e423a0e2c0442da535696a07d112cf
status: failed (33s elapsed)
result_task: get_workflow_details_3
result: 
  failed: true
  return_code: 1
  stderr: "Traceback (most recent call last):
  File "/opt/stackstorm/st2/lib/python3.6/site-packages/st2client/commands/resource.py", line 199, in get_resource_by_pk
    instance = self.manager.get_by_id(pk, **kwargs)
  File "/opt/stackstorm/st2/lib/python3.6/site-packages/st2client/models/core.py", line 42, in decorate
    return func(*args, **kwargs)
  File "/opt/stackstorm/st2/lib/python3.6/site-packages/st2client/models/core.py", line 238, in get_by_id
    self.handle_error(response)
  File "/opt/stackstorm/st2/lib/python3.6/site-packages/st2client/models/core.py", line 195, in handle_error
    response.raise_for_status()
  File "/opt/stackstorm/st2/lib/python3.6/site-packages/requests/models.py", line 943, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 500 Server Error: Internal Server Error
MESSAGE: Internal Server Error for url: http://127.0.0.1:9101/executions/61b26b901838bc652ac0cef3?include_attributes=action.ref%2Caction.runner_type%2Cstart_timestamp%2Cend_timestamp%2Clog%2Cid%2Caction.ref%2Ccontext.user%2Cparameters%2Cstatus%2Cstart_timestamp%2Cend_timestamp%2Clog%2Cresult"
  stdout: 'Action Execution "61b26b901838bc652ac0cef3" is not found.

    ERROR: Resource 61b26b901838bc652ac0cef3 not found.
    '
  succeeded: false
start_timestamp: Thu, 09 Dec 2021 20:48:15 UTC
end_timestamp: Thu, 09 Dec 2021 20:48:48 UTC
log: 
  - status: requested
    timestamp: '2021-12-09T20:48:15.013000Z'
  - status: scheduled
    timestamp: '2021-12-09T20:48:15.122000Z'
  - status: running
    timestamp: '2021-12-09T20:48:15.357000Z'
  - status: failed
    timestamp: '2021-12-09T20:48:48.893000Z'
+--------------------------+------------------------+---------------------------------+------------+-------------------------------+
| id                       | status                 | task                            | action     | start_timestamp               |
+--------------------------+------------------------+---------------------------------+------------+-------------------------------+
| 61b26b8fbcc6a7ed46a5b288 | succeeded (3s elapsed) | execute_inquiry_workflow        | core.local | Thu, 09 Dec 2021 20:48:15 UTC |
| 61b26b93bcc6a7ed46a5b28a | succeeded (1s elapsed) | get_inquiry_trigger             | core.local | Thu, 09 Dec 2021 20:48:19 UTC |
| 61b26b94bcc6a7ed46a5b28c | succeeded (1s elapsed) | get_inquiry_id                  | core.local | Thu, 09 Dec 2021 20:48:20 UTC |
| 61b26b95bcc6a7ed46a5b28e | succeeded (1s elapsed) | get_workflow_id                 | core.local | Thu, 09 Dec 2021 20:48:21 UTC |
| 61b26b96bcc6a7ed46a5b290 | succeeded (2s elapsed) | get_workflow_details_1          | core.local | Thu, 09 Dec 2021 20:48:22 UTC |
| 61b26b98bcc6a7ed46a5b292 | failed (4s elapsed)    | invalid_response_expect_failure | core.local | Thu, 09 Dec 2021 20:48:24 UTC |
| 61b26b9dbcc6a7ed46a5b294 | succeeded (5s elapsed) | pause_after_invalid_response    | core.pause | Thu, 09 Dec 2021 20:48:29 UTC |
| 61b26ba3bcc6a7ed46a5b296 | succeeded (1s elapsed) | get_workflow_details_2          | core.local | Thu, 09 Dec 2021 20:48:35 UTC |
| 61b26ba5bcc6a7ed46a5b2d1 | succeeded (4s elapsed) | valid_response                  | core.local | Thu, 09 Dec 2021 20:48:37 UTC |
| 61b26ba9bcc6a7ed46a5b2d3 | succeeded (6s elapsed) | pause_after_valid_response      | core.pause | Thu, 09 Dec 2021 20:48:41 UTC |
| 61b26bafbcc6a7ed46a5b2d5 | failed (1s elapsed)    | get_workflow_details_3          | core.local | Thu, 09 Dec 2021 20:48:47 UTC |
+--------------------------+------------------------+---------------------------------+------------+-------------------------------+


vagrant@stackstorm:~$ echo "${OUTPUT}" | grep "status" | grep -q "succeeded"
vagrant@stackstorm:~$ echo $?
1
```

This was fun.
You never know where the real demons are :wink: